apis/kueue/v1alpha1/multikueueconfig_types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MultiKueueConfigSpec defines the desired state of MultiKueueConfig
+type MultiKueueConfigSpec struct {
+	// clusters contains the list of configurations for using worker clusters.
+	//
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=100
+	Clusters []MultiKueueCluster `json:"clusters"`
+}
+
+type MultiKueueCluster struct {
+	Name          string        `json:"name"`
+	KubeconfigRef KubeconfigRef `json:"kubeconfigRef"`
+}
+
+type KubeconfigRef struct {
+	SecretName      string `json:"secretName"`
+	SecretNamespace string `json:"secretNamespace"`
+	ConfigKey       string `json:"configKey"`
```

Comment:
Is there a reason this is configurable? The KEP does not let to configure this. I would prefer we have just one hardcoded key (if I'm missing an important use-case then it is ok to keep it).

Reply:
hardcoded now